src/index.ts

```diff
@@ -72,7 +72,21 @@ export const temporal = (<TState>(
         // The order of the get() and set() calls is important here.
         const pastState = options?.partialize?.(get()) || get();
         set(...args);
-        curriedHandleSet(pastState);
+        const currentState = options?.partialize?.(get()) || get();
+        const deltaState = options?.diff?.(pastState, currentState);
+        // Don't call handleSet if state hasn't changed, as determined by equality or diff fn
+        if (
+          !(
+            // If the user has provided an equality function, use it
+            (
+              options?.equality?.(pastState, currentState) ||
+              // If the user has provided a diff function but nothing has been changed, function returns null
+              deltaState === null
+            )
+          )
+        ) {
+          curriedHandleSet(pastState);
```

Comment:
If we go forward with this, what do you think about passing `currentState` and `deltaState` to the `curriedHandleSet`?

```suggestion
          curriedHandleSet(pastState, currentState, deltaState);
```

Reply:
Yeah this makes sense!

Reply:
@charkour Actually I think we may run into some type issues doing this if we want to use a `curriedHandleSet` rather than something more like in PR: 142. I'm not positive and am trying to troubleshoot this now.

The issue is that handleSet is typed as:
```tsx
handleSet?: (
    handleSet: StoreApi<TState>['setState'],
  ) => StoreApi<TState>['setState'];
```

That is, a function that receives a param of type `StoreApi<TState>['setState']`, which is either partial state itself (eg `pastState`) or a state setter fn (I think). Expanding a curried `handleSet` to expect additional params `currentState` and `deltaState` is currently causing some tangles, where we would need to know `currentState` and `deltaState` when we define `curriedHandleState`, which I'm not sure is possible. 

I'll keep thinking about it.

Reply:
Ah, that is tricky, maybe this way isn't doable without a major version change. Could you see what's possible to do without major version changes and then we could consider a future API for v3?

Reply:
Yeah for sure, I'll keep noodling on this and see if I can find a solution that is more of something in between PR 141 and 142, biasing toward something that doesn't change the API

Reply:
Sounds good! Thanks for the help on this!